addons/crm/models/crm_lead.py

```diff
@@ -0,0 +1,1188 @@
+    date_closed = fields.Datetime(string='Closed', readonly=True, copy=False)
+    stage_id = fields.Many2one('crm.stage', string='Stage', track_visibility='onchange', index=True,
+                        domain="['&', ('team_ids', '=', team_id), '|', ('stage_type', '=', lead_type), ('stage_type', '=', 'both')]", default=lambda self: self._default_get_stage_id())
+    user_id = fields.Many2one('res.users', string='Salesperson', index=True, track_visibility='onchange', default=lambda self: self.env.user)
+    referred = fields.Char(string='Referred By')
+    date_open = fields.Datetime(string='Assigned', readonly=True)
+    day_open = fields.Float(compute='_compute_day', string='Days to Assign', store=True)
+    day_close = fields.Float(compute='_compute_day', string='Days to Close', store=True)
+    date_last_stage_update = fields.Datetime(string='Last Stage Update', index=True, default=fields.Datetime.now)
+    date_conversion = fields.Datetime(string='Conversion Date', readonly=True)
+
+    # Messaging and marketing
+    message_bounce = fields.Integer('Bounce', help="Counter of the number of bounced emails for this contact")
+    # Only used for type opportunity
+    probability = fields.Float(group_operator="avg", default=10)
+    planned_revenue = fields.Float(string='Expected Revenue', track_visibility='always')
+    phone = fields.Char(string="Phone", size=64)
+    date_deadline = fields.Date(string='Expected Closing', help="Estimate of the date on which the opportunity will be won.")
+        # CRM Actions
+    last_activity_id = fields.Many2one("crm.activity", string="Last Activity", index=True)
+    next_activity_id = fields.Many2one("crm.activity", string="Next Activity", index=True)
+    next_activity_1 = fields.Char(related="last_activity_id.activity_1_id.name", string="Next Activity 1")
+    next_activity_2 = fields.Char(related="last_activity_id.activity_2_id.name", string="Next Activity 2")
+    next_activity_3 = fields.Char(related="last_activity_id.activity_3_id.name", string="Next Activity 3")
+    date_action = fields.Date(string='Next Activity Date', index=True)
+    title_action = fields.Char(string='Next Activity Summary')
+
+    color = fields.Integer(string='Color Index', default=0)
+    partner_address_name = fields.Char(related='partner_id.name', string='Partner Contact Name', readonly=True)
+    partner_address_email = fields.Char(related='partner_id.email', string='Partner Contact Email', readonly=True)
+    company_currency = fields.Many2one(related='company_id.currency_id', string='Currency', readonly=True, relation="res.currency")
+    user_email = fields.Char(related='user_id.email', string='User Email', readonly=True)
+    user_login = fields.Char(related='user_id.login', string='User Login', readonly=True)
+
+    # Fields for address, due to separation from crm and res.partner
+    street = fields.Char()
+    street2 = fields.Char()
+    zip = fields.Char(change_default=True, size=24)
```

Comment:
keep size attribute here. we can keep size attribute  for this kind of field : code, zip, email, email_*, bic, bank_bic, ifsc,, ifsc_code, iso_code, shortcut.